server/src/controllers/group.controllers.ts

```diff
@@ -0,0 +1,108 @@
+import { Request, RequestHandler, Response } from 'express';
+import { Group } from '../models/group.model';
+import { GroupDocument } from '../types/group.types';
+
+const sendResponse = (
+  res: Response,
+  statusCode: number,
+  payload: GroupDocument[] | GroupDocument | string | null
```

Comment:
is this null for if the error is null? cos i don't see anywhere whereby a null could be returned